README.md

```diff
@@ -203,51 +205,65 @@ Open the project in your IDE to configure the code.
+    ```python
+    @app.route('/a_protected_route')
+    @ms_identity_web.login_required
+    def a_protected_route():
+      return "if you can see this, you're signed in!"
+    ```
 
-```python
-msal_instance = msal.ConfidentialClientApplication(
-    config.get('CLIENT_ID'),
-    client_credential=config.get('CLIENT_SECRET'),
-    authority=config.get('AUTHORITY'),
-    token_cache=None # we don't need a serializable token cache for this project. In-memory token cache will suffice.
-)
-```
-1. The first step of the sign-in process is to send a request to the /authorize endpoint on Azure Active Directory. Our MSAL(Python) ConfidentialClientApplication instance is leveraged to construct an authorization request URL, and our app redirects the browser to this URL.
-1. The user is presented with a sign-in prompt by Azure Active Directory. If the sign-in attempt is successful, the user's browser is redirected to our app's redirect endpoint. A valid request to this endpoint will contain an **authorization code**.
-1. Our ConfidentialClientApplication instance then exchanges this authorization code for an ID Token and Access Token from Azure Active Directory.
+### Under the hood
 
-```python
-token_acquisition_result = msal_instance.acquire_token_by_authorization_code(authorization_code, config.get('SCOPES'))
-```
+In this sample, much of the required MSAL for Python configurations are automatically setup using utilties found in [Microsoft Identity Python Samples Common](https://github.com/azure-samples/ms-identity-python-common). For a more direct, hands-on demonstration of the sign-in process without this abstraction, please see the code within this [Python Webapp](https://github.com/azure-samples/ms-identity-python-webapp) sample.
 
-MSAL Python:
-* Downloads the Azure AD metadata, including signing keys, and finds the issuer name for the tenant.
-* Processes Azure AD responses by validating the signature and issuer in the ID and/or Access tokens.
-* Parses the ID Token claims into plaintext.
- 
-The result is then put into a server-side session, in the following properties:
-```python
-session['msal_token_result']=token_acquisition_result
-session['msal_authenticated']=True
-id_token_claims=token_acquisition_result.get('id_token_claims', {})
-session['msal_id_token_claims'] = id_token_claims
-session['msal_username'] = id_token_claims.get('name', None)
-```
+At a minimum, following parameters need to be provided to MSAL for Python:
+  - The **Client ID** of the app
+  - The **Client Credential**, which is a requirement for Confidential Client Applications
+  - The **Azure AD Authority**, which includes the Tenant ID of the AAD application in this sample's scenario
+
+1. The first step of the sign-in process is to send a request to the `/authorize` endpoint on Azure Active Directory.
+
+1. An MSAL for Python **ConfidentialClientApplication** instance is created:
```

Comment:
>1. An MSAL for Pyth [](start = 0, length = 19)

Mention file name

Reply:
None of this part occurs within the sample code, it's just explaining the flow. It is abstracted by the common code library.

Reply:
added comment explaining that it is in a lib